website/docs/docs/build/udfs.md

```diff
@@ -0,0 +1,152 @@
+
+Note that only basic SQL UDFs are currently supported in dbt.<!--Is this correct?-->
+
+dbt creates, updates, and renames UDFs as part of DAG execution. The UDF file is created before building the model that references it.
+
+## Supported adapters
+
+UDFs are supported in the following adapters:
+
+- BigQuery
+- Snowflake
+- Redshift
+
+## Defining UDFs in dbt
+
+To define UDFs in dbt, refer to the following steps:
+
+1. Create a SQL file under the `functions` directory. For example:
+
+    <File name='functions/is_positive_int.sql'>
+
+    ```sql
+    {{ config(
+        database='udf_db'
+    ) }}
+
+    REGEXP_CONTAINS(a_string, r'^[0-9]+$')
+    ```
+
+    </File>
+
+2. Define your argument, output types, properties, and configs in a corresponding YAML file. For example:
+
+
+    <File name='functions/schema.yml'>
+
+    ```yml
+    functions:
+      - name: is_positive_int # required
+        description: My UDF that determines if a string represents a positive (+) integer # required
+        config:
+            schema: udf_schema
+            database: udf_db
+        arguments: 
+            - name: a_string
+              type: string
+              description: The string that I want to check if it's representing a positive integer (like "10") 
+        return_type: # required
+            type: boolean # required
+    ```
+    </File>
+
+    <Expandable alt_header="Supported UDF types">
+
+    You can use these values for the `type` property when you define a function in a YAML file.
+
+    - `scalar` - Returns a single value per row
+    - `aggregate` - Returns a single value per group, aggregating several rows
+    - `table functions` - Returns a table result
+    <br></br>
+    For example:
+
+    ```yml
+    functions:
+	  - name: string 
+	    description: string
+	    type: scalar # default value
+    ```
+
+    If not explicitly specified, the `type` property defaults to `scalar`.
+
+    </Expandable>
+
+    The rendered `CREATE` UDF statement depends on which adapter you’re using. For example:
+
+    <Tabs>
+
+    <TabItem value="Snowflake">
+    ```sql
+    CREATE OR REPLACE FUNCTION my_schema.is_positive_int(a_string STRING)
+    RETURNS BOOLEAN
+    AS (
+    REGEXP_CONTAINS(a_string, r'^[0-9]+$')
+    );
+    ```
+    </TabItem>
+
+    <TabItem value="Redshift">
+    ```sql
+    CREATE OR REPLACE FUNCTION repeat_n(s VARCHAR, n INTEGER)
+    RETURNS VARCHAR
+    IMMUTABLE
+    AS $$ SELECT repeat(s, n); $$
+    LANGUAGE SQL;
+    ```
+    </TabItem>
+    </Tabs>
+
+3. Reference the UDF in a model using the `{{ function(...) }}` macro. For example:
+
+    <File name="models/my_model.sql">
+
+    ```sql
+    select
+
+    maybe_positive_int_column,
+        {{ function('is_positive_int') }}('maybe_positive_int_column')
```

Comment:
`('maybe_positive_int_column')` shouldn't be in single quotes here. That is, what we pass in to the function _is_ the column, not the name of the column as a string. That is, it should be `(maybe_positive_into_column)` instead.